test/utils.d

```diff
@@ -0,0 +1,211 @@
+module utils;
+
+import vibe.d, std.algorithm, std.process, std.range, std.regex, std.stdio;
+import std.functional, std.string;
+import app;
+
+string testServerURL;
+string ghTestHookURL;
+
+string payloadDir = "./data/payloads";
+string ghHookDir = "./data/hooks";
+
+version(unittest)
+shared static this()
+{
+    // overwrite environment configs
+    githubAuth = "GH_DUMMY_AUTH_TOKEN";
+    hookSecret = "GH_DUMMY_HOOK_SECRET";
+
+    // start our hook server
+    auto settings = new HTTPServerSettings;
+    settings.port = 9000;
+    startServer(settings);
+    startFakeAPIServer();
+
+    testServerURL = "http://" ~ settings.bindAddresses[0] ~ ":"
+                             ~ settings.port.to!string;
+    ghTestHookURL = testServerURL ~ "/github_hook";
+
+    import vibe.core.log;
+    setLogLevel(LogLevel.info);
+
+    runAsync = false;
+    runTrello = false;
+}
+
+void startFakeAPIServer()
+{
+    // start a fake API server
+    auto fakeSettings = new HTTPServerSettings;
+    fakeSettings.port = 9001;
+    fakeSettings.bindAddresses = ["0.0.0.0"];
+    fakeSettings.options = HTTPServerOption.defaults & HTTPServerOption.parseJsonBody;
+    auto router = new URLRouter;
+    router.any("*", &payloadServer);
+
+    listenHTTP(fakeSettings, router);
+
+    auto fakeAPIServerURL = "http://" ~ fakeSettings.bindAddresses[0] ~ ":"
+                                      ~ fakeSettings.port.to!string;
+
+    githubAPIURL = fakeAPIServerURL ~ "/github";
+    trelloAPIURL = fakeAPIServerURL ~ "/trello";
+    travisAPIURL = fakeAPIServerURL ~ "/travis";
+    bugzillaURL = fakeAPIServerURL ~ "/bugzilla";
+}
+
+// serves saved GitHub API payloads
+auto payloadServer(scope HTTPServerRequest req, scope HTTPServerResponse res)
+{
+    import std.path, std.file;
+    APIExpectation expectation = void;
+
+    // simple observer that checks whether a request is expected
+    auto idx = apiExpectations.map!(x => x.url).countUntil(req.requestURL);
+    if (idx >= 0)
+    {
+        expectation = apiExpectations[0];
+        if (apiExpectations.length > 1)
+            apiExpectations = apiExpectations[0 .. idx] ~ apiExpectations[idx + 1 .. $];
+        else
+            apiExpectations.length = 0;
+    }
+    else
+    {
+        assert(0, "Request for unexpected URL received: " ~ req.requestURL);
+    }
+
+    if (expectation.reqHandler !is null)
+        return expectation.reqHandler(req, res);
+
+    string filePath = buildPath(payloadDir, req.requestURL[1 .. $].replace("/", "_"));
```

Comment:
I moved all payloads in one folder and use this `/` -> `_` replacement rule, s.t. one can have a good overview of all available payloads. 
It's still simple to fetch new requests, e.g.

```
api=repos/dlang/dmd/pulls/6328/commits && curl https://api.github.com/${api} > data/payloads/github_$(echo $api | tr '/' '_')
```

However if you prefer to have an exact 1:1 matching with folders, I wouldn't mind either.